sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -692,6 +693,45 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         tasks.RemoveAll(t => t.IsCompleted);
                     }
                 }
+
+                // If the main loop is aborting due to the connection being canceled, then
+                // force the processor to stop.
+                if ((!cancellationToken.IsCancellationRequested) && (Connection.IsClosed))
+                {
+                    Logger.ProcessorClientClosedException(Identifier);
+
+                    // Because this is a highly unusual situation
+                    // and goes against the goal of resiliency for the processor, surface a
+                    // unique exception with explicit message, passing the standard
+                    // ObjectDisposedException as the inner exception.
+                    var fatalException = new ServiceBusException(
+                        false,
+                        Resources.DisposedConnectionMessageProcessorMustStop,
+                        null,
+                        ServiceBusFailureReason.ServiceCommunicationProblem,
+                        new ObjectDisposedException(nameof(ServiceBusConnection)));
+
+                    await InvokeErrorHandler(fatalException, ServiceBusErrorSource.Receive).ConfigureAwait(false);
+
+                    // This call will deadlock if awaited; StopProcessingAsync awaits
+                    // completion of this task.  An exception while stopping will already be logged,
+                    // attach a continuation that will invoke the customer error handler as well.
+                    _ = StopProcessingAsync()
+                        .ContinueWith(async sourceTask =>
+                        {
+                            try
+                            {
+                                await sourceTask.ConfigureAwait(false);
```

Comment:
I don't think there is much value in awaiting this - we expect there to be a bunch of errors since the connection is already closed.

Reply:
We have to get the exception out of the source task here.   We know that it's completed, so we can either pull the `Exception` property and unwrap the aggregate exception manually or await and catch, which has the advantage of unwrapping for us.

Reply:
> We have to get the exception out of the source task here. 

Why? We know things are in a broken state due to the user closing the connection without closing the processor.

Reply:
It seemed inappropriate to have an exception go unobserved by the handler.   Though, you do have a good point that we know things are broken and we've already announced that we have to abort processing.   I'll think on this a bit tonight, but I'm leaning towards your perspective, and that would let us simplify and avoid the continuation.

Reply:
After spinning on it a bit, I'm in agreement.  We're expecting errors and have already signaled the error handler of the root cause.  I've removed the error trap continuation and we'll just run this fire-and-forget;  the actual exceptions will still be logged in the Stop call.